pkg/trace/writer/trace_test.go

```diff
@@ -145,14 +153,23 @@ func randomSampledSpans(spans, events int) *SampledChunks {
 }
 
 // payloadsContain checks that the given payloads contain the given set of sampled spans.
-func payloadsContain(t *testing.T, payloads []*payload, sampledSpans []*SampledChunks) {
+func payloadsContain(t *testing.T, payloads []*payload, sampledSpans []*SampledChunks, shouldUseZstd bool) {
 	t.Helper()
 	var all pb.AgentPayload
 	for _, p := range payloads {
 		assert := assert.New(t)
-		gzipr, err := gzip.NewReader(p.body)
-		assert.NoError(err)
-		slurp, err := io.ReadAll(gzipr)
+		var slurp []byte
+		var err error
+		if shouldUseZstd {
+			zstdReader := zstd.NewReader(p.body)
+			assert.NotNil(zstdReader)
+			slurp, err = io.ReadAll(zstdReader)
+		} else {
+			gzipReader, err := gzip.NewReader(p.body)
+			assert.NoError(err)
+			slurp, err = io.ReadAll(gzipReader)
```

Comment:
This `err` here is the else-branch-scoped `err` from line 168 ( `gzipReader, err := gzip.NewReader(p.body)` ) meaning the below assert won't test it.

Reply:
Fixed thanks a lot.